### ISSUE TITLE
WORKFLOW Notification Mails was not sendet

### ISSUE THREAD

**Issue by hytsch** [joomla/joomla-cms#39505]
### Steps to reproduce the problem

- Create an "Proofreading" group, add your user profile to this group.
- Enable workflow, enable workflow notification emails. 
- Make sure the email sending from the website is working via SMTP.
- Define a workflow phase transition X and enable the email notification option, add the "Proofreading" user group to the recipients.
- Create a post and switch it from phase A to phase B (with transition X).

### Expected result
The notification mail was sent.

### Actual result
The (incorrect) error message appears: "No notifications are sent because there are no users to whom this message can be sent."

### System information (as much as possible)
Joomla 4.2.6 PHP 8.1

### Additional comments

**Comment by chmst** [comment]
Are you the user in the proofreading group and are you the user who performs the transition?
Then the notification is not sent.

**Comment by hytsch** [comment]
Thank you for pointing this out. Now it works. :) 
But I wonder if this behavior is useful. First, the configuration of Notification Mail recipients refers to a freely definable "user group" and there is no indication that someone belonging to this group is excluded from the filter. Moreover, it means that a user can never get the notifications for himself.

**Comment by chmst** [comment]
Same as in forums software or here on github. I get notifications for every post in this repo - but never for mine, seems to be the usual behaviour in notification systems. 
Would some more information in the message help?

**Comment by hytsch** [comment]
Yes, I guess that's an argument. However, one could counter it with the fact that there are quite different types of notifications: Those that inform about an utterance/response in an ongoing conversation (and invite you to reply to a post) and those that only notify about a state or a state change.

The latter are notifications such as notifications about available updates, successfully performed tasks such as backups (Joomla tasks).

In the first case, the notification to the "author" / creator of a post, for example, is only usual, logical, expected in a particular case, since an action by the writer has already taken place. In the second case, however, it can be about general monitoring, just as I as an administrator can subscribe to system messages.

If you ask me, I would want notifications about a state change in a workflow in both cases: If you subscribe yourself as a user to the notification recipients, you should be allowed to receive these notification mails.

**Comment by bembelimen** [comment]
See also https://github.com/joomla/joomla-cms/pull/37860
